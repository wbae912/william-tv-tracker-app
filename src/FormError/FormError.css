.error-message {
  color: rgb(110, 7, 7);
  text-shadow: 0.3px 0.3px rgb(110, 7, 7);
  font-weight: bold;
  margin-top: 5px;
  margin-bottom: 5px;
}