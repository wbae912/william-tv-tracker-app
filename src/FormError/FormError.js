import React from 'react';
import PropTypes from 'prop-types';
import './FormError.css';

function FormError (props) {
    if (props.message) {
        return (
            <div className="error-message">
                {props.message}
            </div>
        );
    }
    return <></>
}

export default FormError;

FormError.propTypes = {
    message: PropTypes.string
}