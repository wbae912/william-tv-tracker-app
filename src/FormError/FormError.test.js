import React from 'react';
import { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import FormError from './FormError';

describe(`FormError component`, () => {
  const message = 'Error message';
  
  it('renders component without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <BrowserRouter>
        <FormError message={message}/>
      </BrowserRouter>
      , div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders a .FormError by default', () => {
    const wrapper = shallow(<FormError message={message}/>)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})