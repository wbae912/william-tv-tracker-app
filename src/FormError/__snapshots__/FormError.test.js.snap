// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormError component renders a .FormError by default 1`] = `
<div
  aria-live="polite"
  className="error-message"
>
  Error message
</div>
`;
